packages/scorchwrap/README.md

```diff
@@ -0,0 +1,145 @@
+# ScorchWrap PoC
+
+> It's like stretchwrap, but with lava
+
+ScorchWrap is a webpack plugin that wraps each module in the bundle in a container and enforces LavaMoat Policies per package.
+
+Implemented capabilities:
+
+ - [x] wrapping built module sources 
+ - [x] runtime to make wrapped modules work
+ - [x] turn off concatenateModules and warn it's incompatible
+ - [x] inlining runtime into the bundle
+ - [x] including actual policies
+ - [ ] using policies to control module loading at runtime
+ - [ ] using policies to limit globals at runtime
+ - [ ] researching and covering all `__webpack_require__.*` functions for security as needed 
+ - [ ] support dynamic imports/requires reaching beyond the bundled content
+ - [ ] paranoid mode checks
+ - [ ] research potential concatenateModules support
+ - [ ] ecosystem compatibility quirks we're yet to find
+
+*(for more specific issues to solve see TODO comments in code)*
+
+
+
+## Usage
+
+The ScorchWrap plugin takes an optional options object with the following properties:
+
+- runChecks: Optional boolean property to indicate whether to check resulting code with wrapping for correctness. Default is false.
+- diagnosticsVerbosity: Optional number property to represent diagnostics output verbosity. A larger number means more overwhelming diagnostics output. Default is 0.  
+  Setting positive verbosity will enable runChecks.
+
+```js
+const ScorchWrapPlugin = require('@lavamoat/scorchwrap');
+
+module.exports = {
+  // ... other webpack configuration properties
+  plugins: [
+    new ScorchWrapPlugin({
+      policy: { ... LavaMoat policy structure ... }
+      runChecks: true,
+    //   diagnosticsVerbosity: 2,
+    }),
+  ],
+  // ... other webpack configuration properties
+};
+```
+
+One important thing to note when using the ScorchWrap plugin is that it disables the `concatenateModules` optimization in webpack. This is because concatenation won't work with wrapped modules.
+
+### Gotchas
+
+- Webpack will gladly add missing dependencies for node builtins like the `events` or `buffer` packages. For LavaMoat policy to handle it, the package needs to be explicitly listed in prod dependencies.
+
+# Security Claims
+
+- SES must be added to the page without any bundling or transforming for any security guarantees to be sustained.
+- Each javascript module resulting from the webpack build is scoped to its package's policy
+
+Threat model
+
+- webpack itself is considered trusted
+- all plugins can bypass LavaMoat protections intentionally
+- It's unlikely but possible that a plugin can bypass LavaMoat protections unintentionally
+- It should not be possible for loaders to bypass LavaMoat protections
+- Some plugins (eg. MiniCssExtractPlugin) execute code from the bundle at build time. To make the plugin work you need to trust it and the modules it runs and add the ScorchWrap.ignore loader for them.
```

Comment:
what if we did some sort of post-bundle analysis?

Reply:
That's part of the scope for paranoid mode. Please write down your ideas if they're more specific than this.